shopfloor/services/zone_picking.py

```diff
@@ -566,6 +591,12 @@ def _scan_source_package(
         package = search.package_from_scan(barcode)
         if not package:
             return response, message
+        if not package.location_id.is_sublocation_of(self.zone_location):
+            # Package is not in an allowed location
+            response = self._list_move_lines(self.zone_location)
+            message = self.msg_store.location_not_allowed()
+            return response, message
+
         move_lines = self._find_location_move_lines(
             locations=sublocation, package=package
```

Comment:
The package has it's own location. We don't need the sublocation anymore.
```suggestion
            package=package
```